lib/beaker/testmode_switcher/beaker_runners.rb

```diff
@@ -58,8 +58,9 @@ def scp_to_ex(from, to)
       end
 
       # execute an arbitrary command on the default host
-      def shell_ex(cmd)
-        shell(cmd)
+      def shell_ex(cmd, opts = {})
+        cmd = "cd #{opts[:chdir]}; " + cmd if opts.key? :chdir
```

Comment:
IIRC beaker ssh currently runs through cygwin, openssh, and bash, so should be no issue. If it becomes one, it needs to be pushed further upstream into beaker, where the different backends can be used to differentiate.

But - as proven by today - that is not a one-afternoon-project.